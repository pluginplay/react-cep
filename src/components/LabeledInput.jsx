import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

class LabeledInput extends React.Component {
  render () {
    return (
      <label className={this.props.className}>
        <span className={'label'}>{this.props.label}</span>
        <span className={'input'}>{this.props.children}</span>
      </label>
    )
  }
}

LabeledInput.propTypes = {
  label: PropTypes.string.isRequired
}

export default styled(LabeledInput)`
  display: flex;
  align-items: center;
  > .input {
    flex-grow: 1;
  }
  > .label {
    margin-right: 5px;
  }
`
