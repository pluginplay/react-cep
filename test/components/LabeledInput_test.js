import { withTheme } from '../setup'
import React from 'react'
import LabeledInput from '../../src/components/LabeledInput'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer
    .create(withTheme(<LabeledInput label={'Look ima label'}><span>foo</span></LabeledInput>))
    .toJSON()
  expect(tree).toMatchSnapshot()
})
