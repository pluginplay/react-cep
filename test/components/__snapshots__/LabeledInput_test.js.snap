// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > .input {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c0 > .label {
  margin-right: 5px;
}

<label
  className="c0"
>
  <span
    className="label"
  >
    Look ima label
  </span>
  <span
    className="input"
  >
    <span>
      foo
    </span>
  </span>
</label>
`;
